res/controllers/common-controller-scripts.js

```diff
@@ -190,6 +192,7 @@ const deepMerge = function(target, source) {
     }
 };
 
+/** @deprecated Use {@link Object.assign} instead */
```

Comment:
This `@deprecated` is not recognized. I think it only works before the function definition.

Reply:
Isn't it? What IDE are you on? This is correctly interpreted on CLion.

Reply:
It neither work with VisualStudio nor VS Code

Reply:
Than it warns in line `script.deepMerge = deepMerge;` that deepMerge is deprecated.

Reply:
My last commit should fix this. Can you test (I don't know how to tell to VSCode "there are global definitions in these files, plz source")?